components/dashboard/src/App.tsx

```diff
@@ -261,32 +264,32 @@ function App() {
         <div className="container">
             <Menu />
             <Switch>
-                <Route path="/new" exact component={NewProject} />
+                <Route path={projectsPathNew} exact component={NewProject} />
                 <Route path="/setup" exact component={Setup} />
-                <Route path="/workspaces" exact component={Workspaces} />
-                <Route path="/account" exact component={Account} />
-                <Route path="/integrations" exact component={Integrations} />
-                <Route path="/notifications" exact component={Notifications} />
-                <Route path="/plans" exact component={Plans} />
-                <Route path="/variables" exact component={EnvironmentVariables} />
-                <Route path="/preferences" exact component={Preferences} />
-                <Route path="/install-github-app" exact component={InstallGitHubApp} />
+                <Route path={workspacesPathMain} exact component={Workspaces} />
```

Comment:
I'm not sure what the advantage is of replacing a simple `<Route path="/literal-path">` with something more complicated like `<Route path={somethingDynamicThatMightFailOneDay}>`. 😬

In the current code, moving, renaming, or adding a new page is pretty simple (requires a single line change).

In the new proposed code, you need to export a new variable somewhere, import it in this file, and use it here (requires a coordinated multi-file change, which is more complicated for the same value).

Maybe I'm missing an important use-case that the more complicated routing allows? (Sorry, haven't read the entire Pull Request yet)

Reply:
@geropl Years ago I used to always have something like `routes.config.ts` with all the routes of the application in a single file. An approach I had borrowed from early releases of Sails.js and it made perfect sense. Like a command center for routes.
But then 4-ish years ago [Nx](https://nx.dev/) changed [how I envision and tend to organise an application](https://egghead.io/courses/scale-react-development-with-nx-4038):
- We can think of each top-level route and component as the head of something we can call "feature" (workspaces, projects, settings, etc are all "features")
- Some of the folders you Gitpodders already organised the code into, tell us you already were thinking in such terms: we already have a folder for "settings", a folder for "workspaces", ...
- Now, what if these features were also units, black boxes, something that works sometimes depending on other units but substantially and primarily by itself. After all, when you visit `/settings` you mostly (only?) do things that are provided by the components in the `settings` folder + some other shared utility components. Nx calls those units "libs" and lets you organise them into actual libs you can lint, test, build, reason about separately... but I digress.
- Given all of the above, having the route paths for a feature in a file inside the feature folder starts making more sense because everything that might happen to those files - moving, renaming, editing, whatever - it will always happen in the same place: the feature folder. And "where are the routes for this feature? Oh right, in the feature folder, of course".
In a word: colocation.

@jankeromnes I'm all for simple magic strings when they are used in just one place.
With the new tests needing the paths to visit and check, the situation changes though.
Having the paths stored in configuration constants:
- enables safe reuse between components and spec files (something we need today)
- enables safe reuse throughout the application wherever you may need to redirect the user from one page to another programmatically or by providing a link:
"ok in this new page X I need to provide a link to the workspaces page... what was the path again? Oh no! I need to check somewhere and find out!" vs "ok in this new page X I need to provide a link to the workspaces page... what was the path again? Easy, I know the convention: feature name, then it's a path, and the main one: `workspacesPathMain` oh, thank you IntelliSense and auto-imports! 🚀"
- protects them from accidental edits (the component files are subject to more frequent changes than a simple configuration file)
- enables robust combination. E.g.:
```ts
export const settingsPathTeams = '/teams';
export const settingsPathTeamsJoin = [settingsPathTeams, 'join'].join('/');
export const settingsPathTeamsNew = [settingsPathTeams, 'new'].join('/');
```
Where you can't accidentally mistype or edit the second and third `/teams` because they literally don't exist, and where what really counts ("join" and "new") is immediately visible
- has the only cost of exporting a const from a file, then you just use it wherever you need it (not just one single file anymore - since we now have test that are using them) by also leveraging the auto-import feature of your favourite IDE

Reply:
> I'm all for simple magic strings when they are used in just one place.
With the new tests needing the paths to visit and check, the situation changes though.

Makes perfect sense to me. :+1: